test/ObserverTest.cpp

```diff
@@ -0,0 +1,153 @@
+#include "pch.h"
+#include <kf/patterns/Observer.h>
+#include <string_view>
+#include <optional>
+
+namespace
+{
+    template<typename... T>
+    class TestObserver : public kf::IObserver<T...>
+    {
+    public:
+        void onNotify(T... t) override
+        {
+            m_values.emplace(std::tuple<T...>(t...));
```

Comment:
Emplace expect parameters for constructor, so it should be:
```suggestion
            m_values.emplace(t...);
```

Reply:
Done